### ISSUE TITLE
no history of previous opam.ocaml.org builds

### ISSUE THREAD

**Issue by avsm** [ocurrent/ocurrent-deployer#190]
<img width="1415" alt="image" src="https://github.com/ocurrent/ocurrent-deployer/assets/53164/b37663fd-d84c-48cb-b8af-f62a02aae03d">

While trying to debug why opam.ocaml.org isn't updating, I see this:
<img width="1350" alt="image" src="https://github.com/ocurrent/ocurrent-deployer/assets/53164/6414db8a-0257-47d4-851d-aab03e49b389">

But no history of builds. Without the history, it's impossible to tell what's causing the build failure. 
@patricoferris noted in another conversation that this may be due to the shape of the ocurrent DAG changing via binds, and so each run of the pipeline is actually a different graph.

In the meanwhile, the deployer is showing orange all the time, and the Hub shows that there hasn't been a push to opam.ocaml.org for 2 days... /cc @tmcgilchrist

**Comment by talex5** [comment]
Note that OCurrent's built-in history only works for builds with the same (primary) key. For example, the history of an ocaml-ci analyse job only shows builds for a single commit (but includes all manual rebuilds and opam-repository updates, where the branch being tested didn't change).

ocaml-ci also has its own web UI, which provides its own history based on the branch name or PR. It would be good to get that supported in OCurrent itself.

**Comment by benmandrew** [comment]
@talex5 I've been very interested in this myself*, but my impression was that OCurrent is application-agnostic and so has no idea about branches or PRs for instance.

I briefly looked at something along the lines of allowing the injection of extra data fields for jobs and search/sort functions into the web UI by a functor, but it seems like a big job.

*It was requested for: 
- [opam-repo-ci](https://github.com/ocurrent/opam-repo-ci) to be able to search by the OCluster server the job is running on
- [multicoretests-ci](https://github.com/ocurrent/multicoretests-ci) to look at the build history of a PR/branch

**Comment by talex5** [comment]
Yes, probably is quite a big job. As well as allowing customisation of the UI, it also needs the code to record the history (e.g. something that combines https://github.com/ocurrent/ocaml-ci/blob/master/lib/index.ml and https://github.com/ocurrent/opam-repo-ci/blob/master/lib/index.ml). Maybe a new sub-library like `current.index` (might need a better name).
